### ISSUE TITLE
running with java 8 nashorn produces 'ReferenceError: "process" is not defined'

### ISSUE THREAD

**Issue by oldag** [requirejs/r.js#976]
using the single page sample app located at https://github.com/volojs/create-template.git, launching r.js with nashorn gives an error:

`tools/r.js:24660 ReferenceError: "process" is not defined`

full text of the session:

```
$ git remote -v
origin	https://github.com/volojs/create-template.git (fetch)
origin	https://github.com/volojs/create-template.git (push)

$ jjs -version
nashorn 1.8.0_161

$ jjs -scripting tools/r.js -- -o tools/build.js
tools/r.js:24660 ReferenceError: "process" is not defined

$ jjs -scripting tools/r.js -- -v
r.js: 2.3.5, RequireJS: 2.3.5, UglifyJS: 2.8.29
```

**Comment by hansbank** [comment]
I faced the same problem. I modified r.js to overcome the problem:
Old code:
`[process.stdout, process.stderr].forEach(function(stream){`
`    if (stream._handle && stream._handle.setBlocking)`
`        stream._handle.setBlocking(true);`
`});`
New code:
`if (env === 'node') {` 
`	[process.stdout, process.stderr].forEach(function(stream){`
`	    if (stream._handle && stream._handle.setBlocking)`
`	        stream._handle.setBlocking(true);`
`	});`
`}`
In other places where process.stdout is used in r.js, there is a condition `env === 'node'`, so I assume it is required here as well.

**Comment by jrburke** [comment]
@hansbank thanks for the pointer on the error. That section should not have made it into the uglifyjs built file, so I removed it in https://github.com/requirejs/r.js/commit/1752638a9b71ce4fc4ac42ce9bcf955ac3c331c8.

I am not sure when a release will be done, but the [current master snapshot](https://raw.githubusercontent.com/requirejs/r.js/master/dist/r.js) includes the change if people want it now.

**Comment by hansbank** [comment]
@jrburke , thanks for the quick fix

**Comment by schubar-tibco** [comment]
Hi @jrburke, thank toy for fixing it. 

It's been 4 months since you have fixed the issue. But because it hasn't been tagged as a release I stumbled upon it in my env. Could you please consider to make a release that includes this fix ([1752638](https://github.com/requirejs/r.js/commit/1752638a9b71ce4fc4ac42ce9bcf955ac3c331c8)).

**Comment by jrburke** [comment]
Sorry for the long delay, but 2.3.6 is released now.
